src/toxcore/tcp/server/server_ext.rs

```diff
@@ -66,21 +86,18 @@ impl ServerExt for Server {
         Box::new(future)
     }
 
-    fn run_connection(self: Self, stream: TcpStream, dht_sk: SecretKey) -> IoFuture<()> {
+    fn run_connection(self: Self, stream: TcpStream, dht_sk: SecretKey) -> Box<Future<Item = (), Error = Error> + Send> {
         let addr = match stream.peer_addr() {
             Ok(addr) => addr,
-            Err(e) => {
-                error!("Could not get peer addr: {}", e);
-                return Box::new(future::err(e))
-            },
+            Err(error) => return Box::new(future::err(ConnectionError::PeerAddrError {
+                error
+            }.into())),
         };
 
         debug!("A new TCP client connected from {}", addr);
 
         let register_client = make_server_handshake(stream, dht_sk.clone())
-            .map_err(|e|
-                Error::new(ErrorKind::Other, format!("Handshake error: {}", e))
-            )
+            .map_err(|e| e.into())
```

Comment:
Should we wrap such errors to `ConnectionError`? `make_server_handshake` will return it's own `failure::Error` so I decided to return it as is. What do you think?

Reply:
Would you try it locally before merge? Will the code compile?

Reply:
It already compiles, but later when `make_server_handshake` returns `failure::Error` we will be able to remove this `map_err`.